File: src/beginner_tutorials/src/gomez_listener.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"

void chatterCallback(const geometry_msgs::Twist::ConstPtr& msg)
{
  if(msg->linear.x > 0) {
      ROS_INFO("GOING FORWARD");
  }
  if(msg->linear.x < 0) {
      ROS_INFO("GOING BACKWARD");
  }
  if(msg->angular.z < 0) {
      ROS_INFO("TURNING RIGHT");
  }
  if(msg->angular.z > 0) {
      ROS_INFO("TURNING LEFT");
  }
}

int main(int argc, char **argv)
{
  
  ros::init(argc, argv, "gomez_listener");

  ros::NodeHandle n;

  ros::Subscriber sub = n.subscribe("cmd_vel", 1000, chatterCallback);

  ros::spin();

  return 0;
}

File: src/beginner_tutorials/scripts/gomez_listener.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import Twist

def callback(data):
    text = ""
    if(data.linear.x > 0): 
        text = "GOING FORWARD"
    elif(data.linear.x < 0):
        text = "GOING BACKWARD"
    elif(data.angular.z > 0):
        text = "TURNING LEFT"
    else:
        text = "TURNING RIGHT"


    rospy.loginfo(rospy.get_caller_id() + text)

def listener():

    rospy.init_node('listener', anonymous=True)

    rospy.Subscriber('cmd_vel', Twist, callback)

    rospy.spin()

if __name__ == '__main__':
    listener()
